### PR TITLE
CCP-1900: changed configurtion for badge image sizing

### PR DESCRIPTION
**Pull request by dqdquint**
In order to be compatible with recently view products and upselling extensions we have to adjust the badge sizing from hard  set width and height values to a width percentage. 
Tested with shop no: 31866 and used configuration found in demo folder.

### CODE REVIEW

**Review by aaron-martin — CHANGES_REQUESTED** [review]
The badge looks too big to me on the product detail page image. Maybe this is how it is meant to look but I think it makes sense to also add a max-width configuration to the configuration. It could be optional and only applied if set in the configuration.  What do you think?
image

**Comment by dqdquint** [comment]
Heyo @aaron-martin , Added a configuration for maximum badge width. Thanks

File: extension-config.json [hunk 1]
```diff
@@ -53,43 +53,24 @@
...
-        "height": 84.5
-      },
+      "default": null,
       "destination": "frontend",
       "params": {
-        "label": "Image ratio for badges within a PDP",
-        "description": "Set height and width to configure a ratio for badges with product images on a PDP",
-        "type": "json"
+        "label": "Maximum Badge Width",
+        "description": "Maximum badge width can be set to px or %",
```
Comment (aaron-martin) [inline extension-config.json]: The sdk reports an error if the % is in the description. When it is out the config file is created without issue.

image

I am not sure it makes sense to allow the max-width to be set as a percent.

**Review by aaron-martin — CHANGES_REQUESTED** [review]

File: frontend/components/Badge/styles.js [hunk 2]
```diff
@@ -1,32 +1,22 @@
...
-  width: `${gridBadgeRatio.width}`,
-  height: `${gridBadgeRatio.height}`,
-}).toString();
-
-const pdpImage = css({
-  width: `${pdpBadgeRatio.width}`,
-  height: `${pdpBadgeRatio.height}`,
+const badge = css({
+  width: `${badgeWidth}%`,
+  maxWidth: `${maxBadgeWidth}`,
```
Comment (aaron-martin) [inline frontend/components/Badge/styles.js]: I don't think maxBadgeWidth needs to be wrapped in a string. I tried it without and it worked fine. Did you find that it had to be wrapped?

**Review by aaron-martin — CHANGES_REQUESTED** [review]

File: frontend/components/Badge/styles.js (hunk 2 above)
Comment (dqdquint) [reply frontend/components/Badge/styles.js]: Ah your write, the config is always a string

**Review by aaron-martin — APPROVED** [review]

**Pull request merged**